plugins/crd/crd_queue.go

```diff
@@ -49,39 +53,53 @@ func networkserviceWork(plugin *Plugin) {
 		// We define a function here to process a queue item, so that we can
 		// use 'defer' to make sure the message is marked as Done on the queue
 		func(key string) {
-			var obj interface{}
-			var err error
-
 			defer queueNS.Done(key)
 
 			// Attempt to split the 'key' into namespace and object name
 			namespace, name, err := cache.SplitMetaNamespaceKey(strKey)
 
 			if err != nil {
+				// This is a soft-error, retry up to QUEUE_RETRY_COUNT times
 				plugin.Log.Errorf("Error splitting meta namespace key into parts: %s", err.Error())
-				runtime.HandleError(fmt.Errorf("Error splitting meta namespace key into parts: %s", err.Error()))
-				// This is a soft-error, we merely want to forget it and mark it done on the queue
-				queueNS.Forget(key)
+				if err != nil {
+					if queueNS.NumRequeues(key) < QUEUE_RETRY_COUNT {
+						plugin.Log.Errorf("Error processing item with key %s, error %v", key, err)
+						queueNS.AddRateLimited(key)
+					} else {
+						plugin.Log.Errorf("Failed processing item with key %s, error %v, no more retries", key, err)
+						queueNS.Forget(key)
+					}
+				}
 				return
 			}
 
 			plugin.Log.Infof("Read item '%s/%s' off workqueue. Processing...", namespace, name)
 
-			// Retrieve the latest version in the cache of this alert
-			obj, err = plugin.sharedFactoryNS.Networkservice().V1().NetworkServices().Lister().NetworkServices(namespace).Get(name)
+			// Retrieve the latest version in the cache of this NetworkService. By using
+			// GetByKey() we are able to determine if the item exists, and thus if it was
+			// added or deleted from the queue, and process appropriately
+			item, exists, err := plugin.informerNS.GetIndexer().GetByKey(strKey)
 
 			if err != nil {
-				plugin.Log.Errorf("Error getting object '%s/%s' from api: %s", namespace, name, err.Error())
-				runtime.HandleError(fmt.Errorf("Error getting object '%s/%s' from api: %s", namespace, name, err.Error()))
-				// This is a hard-error, we'll raise the flag so the plugin catches this and shuts down
-				queueNS.Forget(key)
-				plugin.queueError <- true
```

Comment:
Also, since we are removing this `queueError` functionality here in this PR. There is a goroutine running listening on this `queueError` channel [here](https://github.com/ligato/networkservicemesh/blob/master/plugins/crd/plugin_impl_crd.go#L423) but no one is sending anything to this channel. 

May be that clean up can be done later after this gets in, if we intend to not use it.

Reply:
Good catch, I will remove this.